config/routes.rb

```diff
@@ -1,4 +1,14 @@
 Rails.application.routes.draw do
+  namespace :api do
+    namespace :v1, defaults: { format: :json } do #to specify the default format
+      resources :users do
+        resources :posts, only: [:index] do
+          resources :comments, only: [:index, :create]
+        end
+      end
+    end
+  end
+
   devise_for :users
```

Comment:
- [ ] **Required:** You should also provide a public route for sign-in/log-in just as you have done for sign-out. Without a login route, one cannot log in to get the tokens for authentication. I therefore could not verify the endpoints as authentication must be in place for them to function.

Reply:
@brytebee, thanks for your review.
The login method to get the tokens for authentication has been added
<img width="945" alt="Screen Shot 2022-07-21 at 18 49 24" src="https://user-images.githubusercontent.com/95060298/180333442-884c7a30-0ce1-4953-8c54-c5443e0b112b.png">
.